cmd/common/image_functions.go

```diff
@@ -235,8 +242,37 @@ func doNotAffectDependantManifests(ctx context.Context, manifest acr.ManifestAtt
 	return nil
 }
 
+// getManifestWithoutSubjectToDelete adds the manifest to candidatesToDelete
+// if the manifest does not have subject, otherwise add it to doNotDelete.
+func GetManifestWithoutSubjectToDelete(ctx context.Context, manifest acr.ManifestAttributesBase, doNotDelete set.Set[string], candidatesToDelete []acr.ManifestAttributesBase, acrClient api.AcrCLIClientInterface, repoName string) ([]acr.ManifestAttributesBase, error) {
```

Comment:
Nit. Doing a get function that does an update to a passed in struct is nonstandard and prone to error even if its documented. I'd recommend something like UpdateForManifestWithoutSubjectToDelete which reflects that the passed in values are expected to be modified.